### PR TITLE
Blinking & UI Bars

### PR DESCRIPTION
**Pull request by rezbyte**
[bars.webm](https://user-images.githubusercontent.com/48037381/206138572-db9dab63-6e63-4ec6-aa4a-7216144bfcca.webm)

I've implemented blinking for the player & got the bars on the HUD working:
- Added a generic bar component
- Derived the stamina & blink bars from bar component
- Implemented a blink timer into the player
- Added the `blink` input that allows the player to closes their eyes until released

I didn't know whether you'd prefer these as two separate PRs (that depend on each other) or just one.

### CODE REVIEW

File: Scripts/Player.gd [hunk 1]
```diff
@@ -45,6 +45,14 @@ var closest_interactable: Node3D
 @export var stamina_threshold: float = 0.45
 var stamina_regenerating = false
 
+@export var blink_timer: float = 1.0
+@export var max_blink: float = 1.0
+@export var blink_deplete_rate: float = 0.1
+var blink_current_deplete_rate = blink_deplete_rate
+@onready var blink_overlay: ColorRect = $UI/Blink
```
Comment (dnesov) [inline Scripts/Player.gd]: Player class shouldn't be responsible for UI. Ideally, create a separate node that listens for player's signals and changes it's child controls accordingly.

**Review by dnesov — CHANGES_REQUESTED** [review]
Good job on this PR! However, as I pointed out in Player.gd's comments, we should not have references/logic to UI in this class. I suggest creating a child of Player called "PlayerGUI" or similar that would connect to Player's signals and change the UI whenever Player calls these signals. In the future however, we want to refactor the player script entirely.

**Comment by rezbyte** [comment]
> Good job on this PR! However, as I pointed out in Player.gd's comments, we should not have references/logic to UI in this class. I suggest creating a child of Player called "PlayerGUI" or similar that would connect to Player's signals and change the UI whenever Player calls these signals. In the future however, we want to refactor the player script entirely.

Thanks for the code review! :slightly_smiling_face: 
I've added signals for opening & closing the player's eyes.
As well as signals for changes to stamina & the blink timer.
All of which is handled by a new script attached to the `UI` node in the player.

**Comment by dnesov** [comment]
Much better, thank you for your changes! I'll merge the PR now.

**Pull request merged**